model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/changes/PersistentUpdate.java

```diff
@@ -31,9 +32,11 @@ public class PersistentUpdate {
 
     private final Consumer<KeycloakSession> task;
     private final CompletableFuture<Void> future = new CompletableFuture<>();
+    private final Span span;
 
     public PersistentUpdate(Consumer<KeycloakSession> task) {
         this.task = task;
+        this.span = Span.current();
```

Comment:
Just to clarify things - so it relies on the fact that the "current span" == "span for an associated transaction". Is it correct? So in this context, I believe there will always be some associated transaction, thus always a proper span (when the PR for tracing transactions was merged) for this, right?

Reply:
Yes, as those instances are created during an incoming request there will always be a trace/span.